packages/core/src/browser/xhrObservable.ts

```diff
@@ -0,0 +1,158 @@
+import { callMonitored, monitor } from '../domain/internalMonitoring'
+import { instrumentMethod } from '../tools/instrumentMethod'
+import { Observable } from '../tools/observable'
+import { Duration, elapsed, relativeNow, RelativeTime, ClocksState, clocksNow, timeStampNow } from '../tools/timeUtils'
+import { normalizeUrl } from '../tools/urlPolyfill'
+
+interface BrowserXHR<T> extends XMLHttpRequest {
+  /**
+   * @deprecated this property is shared by both logs and rum and can be used by different code
+   * versions depending on customer setup. To improve reliability and make SDKs independent, this
+   * property should be removed in the future, but this would be a breaking change since some
+   * customers are using it.
+   *
+   * TODO(v4): replace this property with a weakmap index
+   */
+  _datadog_xhr?: T
+}
+
+export interface XhrOpenContext {
+  state: 'open'
+  method: string
+  url: string
+}
+
+export interface XhrStartContext extends Omit<XhrOpenContext, 'state'> {
+  state: 'start'
+  startTime: RelativeTime // deprecated
+  startClocks: ClocksState
+  isAborted: boolean
+  xhr: XMLHttpRequest
+}
+
+export interface XhrCompleteContext extends Omit<XhrStartContext, 'state'> {
+  state: 'complete'
+  duration: Duration
+  status: number
+  responseText: string | undefined
+}
+
+export type XhrContext = XhrOpenContext | XhrStartContext | XhrCompleteContext
+
+let xhrObservable: Observable<XhrContext> | undefined
+
+export function initXhrObservable() {
+  if (!xhrObservable) {
+    xhrObservable = createXhrObservable()
+  }
+  return xhrObservable
+}
+
+function createXhrObservable() {
+  const observable = new Observable<XhrContext>(() => {
+    const { stop: stopInstrumentingStart } = instrumentMethod(
+      XMLHttpRequest.prototype,
+      'open',
+      (original) =>
+        function (method, url) {
+          callMonitored(openXhr, this, [method, url])
+          return original.apply(this, arguments as any)
+        }
```

Comment:
This pattern seems to be quite used, so we may have a helper function to factorize it, ex:

```ts

const { stop: stopInstrumentingStart } = instrumentMethodAndCallOriginal(
      XMLHttpRequest.prototype,
      'open',
      openXhr
)
```

The only thing is that it doesn't allow to modify the arguments, which is needed for tracking fetch., so we need to keep the current `instrumentMethod` implementation